### PR TITLE
Cucumber3

### PR DESCRIPTION
**Pull request by jaffamonkey**
Updates from master, and additional steps to cover address-related features

### CODE REVIEW

File: tests/src/features/0-user-journey-primary-authority-new-direct-partnership.feature [hunk 1]
```diff
@@ -1,45 +1,21 @@
...
-
-        # CHOOSE AUTHORITY REPRESENTED
-
         Then I expect that element "h1.heading-xlarge" contains the text "Which authority are you acting on behalf of?"
         And I click on the radio "label*=Metropolitan District"
         And I click on the button "#edit-next"
-
-        # CHOOSE PARTNERSHIP TYPE
-
-        Then I expect that element "h1.heading-xlarge" contains the text "New partnership application"
```
Comment (stemount) [inline tests/src/features/0-user-journey-primary-authority-new-direct-partnership.feature]: Let's try get these text assertions back in, they're useful for when we change other parts of the site that affect other forms/views!

File: tests/src/features/0-user-journey-primary-authority-new-direct-partnership.feature [hunk 2]
```diff
@@ -1,45 +1,21 @@
...
         When I click on the radio "#edit-application-type-direct"
         And I click on the button "#edit-next"
-
-        # CONFIRMATIONS
-
         When I click on the checkbox "#edit-business-eligible-for-partnership"
-        And I click on the button "#edit-next"
```
Comment (stemount) [inline tests/src/features/0-user-journey-primary-authority-new-direct-partnership.feature]: these could stay, this should not have changed, if they have, this is a regression :)

File: tests/src/features/00-user-journey-partnership-revocation.feature [hunk 3]
```diff
@@ -29,35 +36,38 @@ Feature: New Direct Partnership For HD user To Revoke
...
-        When I add "London" to the inputfield "#edit-town-city"
-        When I add "London" to the inputfield "#edit-county"
-        And I select the option with the text "England" for element "#edit-nation"
+        And I run tota11y against the current page
+        And I add "Leidseplein 10" to the inputfield "#edit-address-line1"
+        And I add "Leidseplein" to the inputfield "#edit-address-line2"
+        When I add "Amsterdam" to the inputfield "#edit-town-city"
+        Then I expect that element "#edit-country-code" is visible
+        And I select the option with the text "Netherlands" for element "#edit-country-code"
+        When I add "1017 PT" to the inputfield "#edit-postcode"
```
Comment (stemount) [inline tests/src/features/00-user-journey-partnership-revocation.feature]: nice :) 🇳🇱

File: tests/src/support/action/tota11y.js [hunk 4]
```diff
@@ -12,7 +12,9 @@ module.exports = () =>
         browser.click('.tota11y-plugin-title*=' + list[i]);
         if (browser.isVisible('.tota11y-info-errors') === true) {
             const errors = browser.getText('.tota11y-info-errors');
-            expect(browser.isVisible('.tota11y-info-errors')).to.not.equal(true, errors);
+            var url = browser.getUrl();
+            console.log(url, errors);
```
Comment (stemount) [inline tests/src/support/action/tota11y.js]: tested that this prints to the travis log 👍

File: tests/src/features/0-user-journey-primary-authority-new-direct-partnership.feature (hunk 2 above)
Comment (kalpaitch) [reply tests/src/features/0-user-journey-primary-authority-new-direct-partnership.feature]: It's checking the validation of the elements, agree that we still need this test, at least somewhere.

File: tests/src/features/00-user-journey-partnership-revocation.feature [hunk 5]
```diff
@@ -29,36 +36,43 @@ Feature: New Direct Partnership For HD user To Revoke
...
 
         # ADD BUSINESS DETAIL
-
-        When I add "SE16 4NX" to the inputfield "#edit-postcode"
-        And I add "1 Change St" to the inputfield "#edit-address-line1"
-        And I add "New Change" to the inputfield "#edit-address-line2"
-        When I add "London" to the inputfield "#edit-town-city"
-        When I add "London" to the inputfield "#edit-county"
-        And I select the option with the text "England" for element "#edit-nation"
+        And I run tota11y against the current page
```
Comment (kalpaitch) [inline tests/src/features/00-user-journey-partnership-revocation.feature]: Ideally we'd test against London as well, I assume we're still doing this in another test right?

**Pull request merged**